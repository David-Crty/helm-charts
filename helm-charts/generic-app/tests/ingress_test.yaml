# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json

suite: test ingress
release:
  name: test
chart:
  version: 0.1.0
templates:
  - ingress.yaml
tests:
  - it: should have ingress
    set:
      ingress:
        enabled: true
        hosts:
          - test.awstudio.cloud
    asserts:
      - matchSnapshot: {}
  - it: should ingress with keycloak middleware annotation
    set:
      ingress:
        enabled: true
        keycloakProtection: true
    asserts:
      - matchSnapshot: {}