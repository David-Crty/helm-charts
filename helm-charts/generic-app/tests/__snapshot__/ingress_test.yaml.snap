should create middleware redirections:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: letsencrypt-prod
        traefik.ingress.kubernetes.io/router.middlewares: traefik-test-generic-app-redirect-to-crty-dev@kubernetescrd,traefik-redirect-to-https@kubernetescrd
      name: test-generic-app
    spec:
      ingressClassName: traefik
      rules:
        - host: crty.dev
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
        - host: www.crty.dev
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
        - host: www.crty.com
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
        - host: crty.com
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - crty.dev
            - www.crty.dev
            - www.crty.com
            - crty.com
          secretName: test-generic-app-cert
  2: |
    apiVersion: traefik.io/v1alpha1
    kind: Middleware
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: generic-app
        helm.sh/chart: generic-app-0.1.0
      name: test-generic-app-redirect-to-crty-dev
      namespace: traefik
    spec:
      redirectRegex:
        permanent: true
        regex: ^https?://(www\.crty\.dev|www\.crty\.com|crty\.com)(/.*)
        replacement: https://crty.dev$2
should have ingress:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: letsencrypt-prod
        traefik.ingress.kubernetes.io/router.middlewares: traefik-redirect-to-https@kubernetescrd
      name: test-generic-app
    spec:
      ingressClassName: traefik
      rules:
        - host: test.awstudio.cloud
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - test.awstudio.cloud
          secretName: test-generic-app-cert
should ingress with keycloak middleware annotation:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: letsencrypt-prod
        traefik.ingress.kubernetes.io/router.middlewares: traefik-redirect-to-https@kubernetescrd,traefik-keycloakopenid@kubernetescrd
      name: test-generic-app
    spec:
      ingressClassName: traefik
      rules:
        - host: hello-world.com
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - hello-world.com
          secretName: test-generic-app-cert
